pages/transactions/index.tsx

```diff
@@ -9,19 +9,121 @@ import { Button,
    FormControl,
    InputLabel,
    Select,
-   MenuItem}
+   MenuItem,
+   Popover,
+   TextField,
+   RadioGroup,
+   FormControlLabel,
+   Radio,
+   Accordion,
+   AccordionSummary,
+   AccordionDetails,
+   Autocomplete,}
 from "@mui/material";
 import styles from './Transactions.module.css';
-import itemstyles from '../../components/TransactionItem/TransactionItem.module.css';
 import { TabPanel } from '../../components/TabPanel/TabPanel';
 import Icon from '../../assets/Icon';
-import { Transaction } from '../../types/schema';
 import { TransactionList } from '../../components/TransactionList/TransactionList';
-import { getAllTransactions } from '../../firebase/firestore/transaction'; 
-import { getAdmin } from '../../firebase/firestore/admin';
-import { getUser } from '../../firebase/firestore/user';
+import { getAllTransactions, addTransaction } from '../../firebase/firestore/transaction'; 
+import { getAllUsers } from '../../firebase/firestore/user';
+import { Transaction } from '../../types/schema';
  
 const TransactionsPage: React.FunctionComponent = () => {
+    const [addAnchorEl, setAddAnchorEl] = React.useState(null);
+    const [uploadAnchorEl, setUploadAnchorEl] = React.useState(null);
+    const [allUsers, setUsers] = React.useState([]);
+    const [allTransactions, setTransactions] = useState([]);
+    const [success, setSuccess] = useState(false);
+    const [selectedUser, setSelectedUser] = useState('Select User')
+    const [expanded, setExpanded] = useState(false);
+    const [addUser, setAddUser] = useState(null);
+    const [addPoints, setAddPoints] = useState('10');
+    const [addType, setAddType] = useState('');
+    const [addMessage, setAddMessage] = useState('');
+
+    useEffect(() => {
+        getAllUsers().then(users => {
+            setUsers(users);
+        })
+        getAllTransactions().then(items => {
+            setTransactions(items);
+        })
+    }, []);
+
+    /*
+    added this sleep function because handleClose function set success back to false before the anchor was set to null
+    (meaning that the) popover would switch back to the first page before closing.
+    The sleep makes this look cleaner, but let me know if there is a less hacky fix.
+    */
+    const sleep = (ms) => {
```

Comment:
Could you actually explain in a bit more detail why this sleep is necessary?

Reply:
it's easier if i show u but there are two main variables that handle the add popover: addAnchorEl, which handles whether the popover is open or not, and success, which decides whether the content on the popover is the form or the success screen.

When we hit close on the success screen, we need to set addAnchorEl to null, which will close the popover. Then we also need to set success back to false so that when we open the popover next, it'll show the form, not the success screen.

The issue that I encountered here was that success would be set to false before addAnchorEl was set to null, meaning that we would see the form reappear before the popover disappeared. My hacky fix is therefore to create a bit of lag time so that success would not happen first. I feel like there should be a better way with .then but idk what it is